app/models/event.rb

```diff
@@ -13,7 +13,12 @@ class Event < ActiveRecord::Base
   has_many :categories, through: :categorizations
   has_many :items
 
-  scope :active,      -> { joins(:items).merge(Item.available).open_events }
+  scope :active,      -> { joins(:items).uniq.merge(Item.available).open_events }
   scope :open_events, -> { where("date >= ?", Date.today).is_approved }
   scope :is_approved, -> { where approved: true }
+
+  def self.active_events(venue)
+    active.where("venue_id = ?", venue.id)
+  end
```

Comment:
see note above, about possibly using the scopes instead of making a new class method.